Source/ACE.Database/CharacterDatabase.cs

```diff
@@ -133,13 +136,60 @@ public uint GetMaxId()
             }
             else
             {
+                CharacterPosition newCharacterPosition = CharacterPositionExtensions.StartingPosition;
+                newCharacterPosition.character_id = id;
+                newCharacterPosition.positionType = (int)CharacterPositionType.PhysicalLocation;
+
+                // Did not find a position in the database, so we will create one here.
+                ExecuteConstructedInsertStatement(CharacterPreparedStatement.CharacterPositionUpdate, typeof(CharacterPosition), newCharacterPosition);
+
                 // use fallback position if position information doesn't exist in the DB, show error in the future
                 pos = new Position(0x7F0401AD, 12.3199f, -28.482f, 0.0049999995f, 0.0f, 0.0f, -0.9408059f, -0.3389459f);
             }
 
             return pos;
         }
 
+        /// <summary>
+        /// Returns the Player's CharacterPosition for the type provided the paramaters
+        /// </summary>
+        public async Task<CharacterPosition> GetCharacterPosition(uint id, CharacterPositionType type)
+        {
+            // Query for certain type
+            MySqlResult result = await SelectPreparedStatementAsync(CharacterPreparedStatement.CharacterPositionSelect, id, type);
+            CharacterPosition newCharacterPosition = new CharacterPosition();
+            if (result.Count > 0)
+            {
+                newCharacterPosition.cell = result.Read<uint>(0, "cell");
```

Comment:
That was updated from a previous slide; I've pushed a commit with updated ORM calls.